### PR TITLE
#379 review home part 3

### PR DESCRIPTION
**Pull request by andreievg**
Fixes; #379 
Branched from: #379-review-home-part-2

### Changes

* noticed that the same error was being use (application overview) in a lot of pages, changed it to generic
* useGetFullReviewStructure to also store reviewQuestionAssignmentId in pageElement
* added method to useCreateReview `createReviewFromStructure` to do all of the logic of creating a review
* added StartReviewButton to `ReviewSectionRowAction`

### CODE REVIEW

File: src/utils/constants.ts [hunk 1]
```diff
@@ -38,6 +38,7 @@ export default {
   ERROR_APPLICATION_SUBMIT: 'Problem submitting application',
   ERROR_APPLICATIONS_LIST: 'Problem loading applications list',
   ERROR_REVIEW_APPLICATION: 'Problem loading application for review',
+  ERROR_GENERIC: 'Something went wrong',
   ERROR_REVIEW_PAGE: 'Problem loading review',
   ERROR_REVIEW_OVERVIEW: 'Problem loading review homepage',
   ERROR_REVIEW_SECTION_ROW: 'Problem loading review section row',
```
Comment (nmadruga) [inline src/utils/constants.ts:44]:
```suggestion
  ERROR_GENERIC: 'Something went wrong',
  ERROR_REVIEW_PAGE: 'Problem loading review',
  ERROR_REVIEW_OVERVIEW: 'Problem loading review homepage',
```

Other error strings to be replaced by Generic error (and removed from here) as part of changes done in #369 - Already added a comment to remember about doing this in there)

File: src/utils/hooks/useCreateReview.tsx [hunk 2]
```diff
@@ -40,7 +46,24 @@ const useCreateReview = ({ reviewerId, serialNumber, onCompleted }: UseCreateRev
...
+
+    return await createReviewMutation({
+      variables: {
+        reviewAssigmentId,
+        applicationResponses,
+      },
+    })
+  }
+
+  const createReview = ({ applicationResponses }: CreateReviewProps) => {
```
Comment (nmadruga) [inline src/utils/hooks/useCreateReview.tsx:66]: This is here only for existing use? Good to leave a comment for this to be removed. I suggest:
`// TODO: Remove this`

If you do a global search there are many places with this comment to help us when deleting all the old stuff

**Review by nmadruga — COMMENTED** [review]
Looks good. Just a few comments to have a look please

File: src/utils/constants.ts (hunk 1 above)
Comment (andreievg) [reply src/utils/constants.ts:44]: I've adde a comment, here; https://github.com/openmsupply/application-manager-web-app/issues/369#issuecomment-800704948, basically i noticed wrong error string was being used, and changed to generic (didn't really want to change all of them to correct ones, i think we should still use specific errors)

File: src/components/Review/ReviewSectionRowAction.tsx [hunk 3]
```diff
@@ -45,4 +50,45 @@ const ReviewSectionRowAction: React.FC<ReviewSectionComponentProps> = ({
...
+    push,
+  } = useRouter()
+
+  const [startReviewError, setStartReviewError] = useState(false)
+
+  const { createReviewFromStructure } = useCreateReview({
+    reviewAssigmentId: assignment.id,
+    reviewerId: currentUser?.userId as number,
+    serialNumber: fullStructure.info.serial,
+    onCompleted: () => {},
```
[the remark this answers is not in the document]
Comment (andreievg) [reply src/components/Review/ReviewSectionRowAction.tsx:73]: i think it's still used (in previous way), i've added a todo cd40cf8f3145b29aa37e8877578b52d05bd361d8

File: src/utils/hooks/useCreateReview.tsx (hunk 2 above)
Comment (andreievg) [reply src/utils/hooks/useCreateReview.tsx:69]: cd40cf8f3145b29aa37e8877578b52d05bd361d8

File: src/utils/constants.ts (hunk 1 above)
Comment (nmadruga) [reply src/utils/constants.ts:43]: Oh! I think I haven't get the idea correctly sorry! I'll ament to my comment on the other issue

Comment (nmadruga) [reply src/utils/constants.ts:43]: You can already remove not in use strings `ERROR_REVIEW_APPLICATION` and `ERROR_REVIEW_SECTION_ROW`. But not a big requirement anyway

**Review by nmadruga — APPROVED** [review]
LGTM : )

File: src/utils/constants.ts (hunk 1 above)
Comment (andreievg) [reply src/utils/constants.ts:43]: I was wondering if we can continue that discussion after lunch, and sum it up here: https://github.com/openmsupply/application-manager-web-app/issues/369

**Comment by andreievg** [comment]
Thanks for review Nicole

**Pull request merged**